test/e2e/tests/commons/sinspqa/sinsp.py

```diff
@@ -0,0 +1,218 @@
+from datetime import datetime
+from time import sleep
+import os
+import json
+import docker
+import re
+from enum import Enum
+
+SINSP_TAG = os.environ.get('SINSP_TAG', 'latest')
+
+
+class SinspStreamer:
+    """
+    Allows streaming of `sinsp-example` logs for analysis.
+    """
+
+    def __init__(self, container, timeout=10):
+        """
+        Parameters:
+            container (docker.Container): A container object to stream logs from.
+            timeout (int): The maximum amount of time the streamer will read logs from the container.
+        """
+        self.container = container
+        self.timeout = timeout
+        self.last_timestamp = None
+
+    def read(self):
+        """
+        Reads logs from a container and returns them as a generator.
+
+        Returns:
+            A string holding a single log line from the container.
+        """
+        start = datetime.now()
+
+        while True:
+            sleep(0.2)
```

Comment:
The problem with `container.logs(stream=True)` is that it is blocking and doesn't return control to the calling thread until the container stops, because the `sinsp-example` only stops when it receives a signal we completely lose the ability to do anything on the main thread. I might be something wrong though, I'll try to find a better way to do this.

Reply:
Ok, my previous comment was wrong, `follow=True` had the described behavior. I've had a ton of problems trying to get the docker logs to be properly streamed into the tests, this one has a few repeated lines every time the EOF is reached and `docker.logs` is called again which wouldn't cause problems unless we start looking for consecutive repeated events.